src/plugins/region_map/server/plugin.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+} from 'opensearch-dashboards/server';
+
+import { OpenSearchService } from './services';
+import { opensearch } from '../server/routes';
+import { getUiSettings } from './ui_settings';
+
+export class RegionMapPlugin implements Plugin<object, object> {
+  private readonly logger: Logger;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+  }
+
+  public setup(core: CoreSetup) {
+    this.logger.debug('RegionMap: Setup');
+    const opensearchClient = core.opensearch.legacy.createClient('opensearch');
```

Comment:
Will there be a follow up to use the non legacy client or is there a specific reason for using legacy?

Reply:
For this release it will be great if we could keep the legacy client. I referred this https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/index_pattern_management/server/routes/resolve_index.ts, but we can change it in the future release, what say @kavilla ?

Reply:
Sorry, I'm not quite following - why did we choose the legacy client?

Reply:
I just followed the pattern used in the mentioned plugin : ) I wasn't sure if I had a choice. What is the other way of achieving the same thing?

Reply:
Removed legacy client from region map plugin.